ground/src/test/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragmentTest.kt

```diff
@@ -0,0 +1,109 @@
+ */
+package com.google.android.ground.ui.offlineareas
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.R
+import com.google.android.ground.launchFragmentInHiltContainer
+import com.google.android.ground.persistence.local.stores.LocalOfflineAreaStore
+import com.google.android.ground.util.drwable.withDrawable
+import com.google.android.ground.util.recyclerview.atPositionOnView
+import com.sharedtest.FakeData.OFFLINE_AREA
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import kotlinx.coroutines.test.advanceUntilIdle
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class OfflineAreasFragmentTest : BaseHiltTest() {
+
+  lateinit var fragment: OfflineAreasFragment
+  @Inject lateinit var localOfflineAreaStore: LocalOfflineAreaStore
+
+  @Before
+  override fun setUp() {
+    super.setUp()
+    launchFragmentInHiltContainer<OfflineAreasFragment> { fragment = this as OfflineAreasFragment }
+  }
+
+  @Test
+  fun `toolbar text should be correct`() {
+    onView(withId(R.id.offline_areas_toolbar))
+      .check(matches(hasDescendant(withText(fragment.getString(R.string.offline_map_imagery)))))
+  }
+
+  @Test
+  fun `heading should be correct`() {
+    onView(withId(R.id.offline_areas_list_title))
+      .check(matches(withText(fragment.getString(R.string.offline_downloaded_areas))))
+    onView(withId(R.id.offline_areas_list_tip))
+      .check(matches(withText(fragment.getString(R.string.offline_area_list_tip))))
+    onView(withId(R.id.no_areas_downloaded_message))
+      .check(matches(withText(fragment.getString(R.string.no_basemaps_downloaded))))
+  }
+
+  @Test
+  fun `list is visible`() = runWithTestDispatcher {
+    localOfflineAreaStore.insertOrUpdate(OFFLINE_AREA)
+
+    advanceUntilIdle()
+    onView(withId(R.id.offline_areas_list)).check(matches(isDisplayed()))
+
+    verifyTextOnOfflineAreasListItemAtPosition(
+      itemPosition = 0,
+      targetView = R.id.offline_area_list_item_name,
+      stringToMatch = "Test Area",
+    )
+    verifyTextOnOfflineAreasListItemAtPosition(
+      itemPosition = 0,
+      targetView = R.id.offline_area_list_item_size,
+      stringToMatch = "<1\u00A0MB",
+    )
+    onView(
+        atPositionOnView(
+          recyclerViewId = R.id.offline_areas_list,
+          position = 0,
+          targetViewId = R.id.offline_area_list_item_icon,
+        )
+      )
+      .check(matches(withDrawable(R.drawable.ic_offline_pin)))
```

Comment:
I don't see any values in validating that this drawable is present since it is always present regardless of the item

Reply:
The purpose of the test shouldn't be to verify if android's core API is working or not. If for some reason the drawable doesn't load on certain android versions, then this unit test won't be able to catch those anyways. We can keep the util class you created for comparing the drawable if you feel it's useful but I don't feel it adds too much value to this test case.

Reply:
correct, got your point, let's keep the util stuff ad removing this test case